static/js/chart/draw.ts

```diff
@@ -724,6 +735,80 @@ function drawLineChart(
         dots.on("click", handleDotClick);
       }
     }
+
+    if (highlightOnHover) {
+      addTooltip(id);
+      highlight.style("opacity", "0");
+      const highlightedDot = highlight
+        .append("circle")
+        .attr("r", 5)
+        .style("fill", colorFn(dataGroup.label))
+        .style("stroke", "#fff");
+      const highlightedLine = highlight
+        .append("line")
+        .attr("x1", 0)
+        .attr("y1", 0)
+        .attr("x2", 0)
+        .attr("y2", height - MARGIN.bottom)
+        .style("stroke", colorFn(dataGroup.label))
+        .style("stroke-opacity", "0.5");
+      const tooltip = d3.select(`#${id}`).select(`#${TOOLTIP_ID}`);
+      svg
+        .on("mouseover", () => {
+          highlight.style("opacity", "1");
+          tooltip.style("display", "block");
+        })
+        .on("mouseout", () => {
+          highlight.style("opacity", "0");
+          tooltip.style("display", "none");
+        })
+        .on("mousemove", () => {
+          // calculation of data point corresponding to mouse position from https://bl.ocks.org/Qizly/8f6ba236b79d9bb03a80.
+          const mouseX = d3.mouse(d3.select("#" + id).node() as HTMLElement)[0];
+          const highlightedTime = xScale.invert(mouseX).getTime();
+          const bisect = d3.bisector((d) => d[0]);
+          let idx = bisect.right(dataset, highlightedTime);
+          if (idx > 0 && idx < dataset.length) {
+            idx =
+              dataset[idx][0] - highlightedTime >
+              highlightedTime - dataset[idx - 1][0]
+                ? idx - 1
+                : idx;
+          } else if (idx === dataset.length) {
+            idx = idx - 1;
+          }
+          if (dataset[idx]) {
+            const datapointX = xScale(dataset[idx][0]);
+            const datapointY = yScale(dataset[idx][1]);
+            highlightedDot.attr(
+              "transform",
+              "translate(" + datapointX + "," + datapointY + ")"
+            );
+            highlightedLine.attr(
+              "transform",
+              "translate(" + datapointX + "," + "0)"
+            );
+            const dataPoint: DataPoint = dataGroup.value[idx];
+            const tooltipText = `${dataPoint.label}: ${dataPoint.value} ${
+              unit ? unit : ""
```

Comment:
for place + timeline explorer, this would need to use the i18n functions. could you please add a todo for that?

Reply:
done